### PR TITLE
Document aimed to describe OF cookies format

### PR DESCRIPTION
**Pull request by surabujin**

### CODE REVIEW

**Review by timofei-durakov — APPROVED** [review]

**Review by niksv — APPROVED** [review]

File: docs/design/multi-table-pipelines/cookies.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+^                                                                     ^
+`- 63 bit                                                             `- 0 bit
+```
+
+* S - SERVICE_FLAG (1 bit) - OF flow marked with 1 in this bit is a service flow, required for open-kilda to manage service traffic (ISL detection / health check, loops prevention etc)
+* T - TYPE_FIELD (9 bit) - contain cookie type code, this type defines existence and layout all other bit fields in the cookie
+
+
+## Generic cookies (org.openkilda.model.cookie.CookieBase)
+This cookie format was the only existing format into open-kilda from its start. It is not put any restrictions on cookie format and can be used to read "raw" cookie value to determine its actual type.
```
Comment (rozdy) [inline docs/design/multi-table-pipelines/cookies.md:19]: `It is not put any restrictions` -> `It doesn't put any restrictions` or `It put no restrictions`

File: docs/design/multi-table-pipelines/cookies.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+## Kilda-flow's segment cookies (org.openkilda.model.cookie.FlowSegmentCookie)
+Fields:
+
+```
+_FR_____ ____L000 00000000 00000000 00000000 0000IIII IIIIIIII IIIIIIII
+^                                                                     ^
+`- 63 bit                                                             `- 0 bit
+```
+
+* I - FLOW_EFFECTIVE_ID_FIELD (20 bits) - unique numeric kilda-flow identifier (equal across all path)
```
Comment (rozdy) [inline docs/design/multi-table-pipelines/cookies.md:45]: across all flow paths

File: docs/design/multi-table-pipelines/cookies.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+* p - PORT_NUMBER_FIELD (16 bits) - number of switch port with OF flow belongs to
+* V - VLAN_ID_FIELD (12 bits) - vlanId this OF flow matches
+
+Constraints:
+
+* SERVICE_FLAG == 0
+* TYPE_FIELD == SHARED_OF_FLOW(0x008)
+
+Possible values of shared segments:
+* QINQ_OUTER_VLAN == 0
```
Comment (rozdy) [inline docs/design/multi-table-pipelines/cookies.md:75]: Didn't get it. How it changes the cookie?

File: docs/design/multi-table-pipelines/cookies.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+```
+_FR_____ ____0000 00000000 00000000 00000000 0000EEEE EEEEEEEE EEEEEEEE
+^                                                                     ^
+`- 63 bit                                                             `- 0 bit
+```
+
+* E - EXCLUSION_ID_FIELD (20 bits) - unique exclusion numeric identifier 
+* R - FLOW_REVERSE_DIRECTION_FLAG (1 bit) - refer to kilda-flow's reverse path (Z-to-A) 
+* F - FLOW_FORWARD_DIRECTION_FLAG (1 bit) - refer to kilda-flow's forward path (Z-to-A)
```
Comment (rozdy) [inline docs/design/multi-table-pipelines/cookies.md:111]: A-to-Z

**Review by rozdy — APPROVED** [review]

**Pull request merged**